modules/bootconf.py

```diff
@@ -118,3 +127,94 @@ def configure(self, old, new):
             return False
 
         return True
+
+class UBootBeagleboneBootloader(BootloaderConfigurator):
+
+    def switchUEnv(self, old, new, uEnvPath):
+        ''' Switch boot partition using uEnv bootpart variable '''
+        oldidx = getPartitionIndex(old)
+        newidx = getPartitionIndex(new)
+        if not oldidx or not newidx:
+            return False
+
+        if not super(UBootBeagleboneBootloader, self).applyTextTransformation(uEnvPath, 'bootpart=1:' + oldidx, 'bootpart=1:' + newidx):
+            return False
+
+        return True
+
+    def tweakUEnv(self, bootmountpoint):
+        ''' Various tweaks and cleanups to uEnv.txt '''
+
+        log.info ("tweakUEnv: Tweaking and cleaning up uEnv.txt...")
+
+        if not isMounted(bootmountpoint):
+            log.error("transformUEnv: " + bootmountpoint + " is not a mountpoint.")
+            return False
+        uEnvPath = os.path.join(bootmountpoint, 'uEnv.txt')
+        if not os.path.isfile(uEnvPath):
+            log.error("transformUEnv: uEnv.txt seem not to exist in boot partition")
+            return False
+
+        lines=[]
+        fixFindUUID = True
```

Comment:
Sadly it is. This is because it greatly depends on how the default environment is created in uboot. And in this case finduuid is defined in ti_armv7_common.h which is TI specific for armv7 boads.